### PR TITLE
CC-31652 Introduced `datasource.dependable` and `datasource.trigger` services.

### PR DESCRIPTION
**Pull request by danilmoro3**
Ticket: https://spryker.atlassian.net/browse/CC-31652

### CODE REVIEW

**Review by FRITS-Kh — CHANGES_REQUESTED** [review]

**Review by FRITS-Kh — APPROVED** [review]

File: docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-dependable.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    link: docs/scos/dev/front-end-development/page.version/marketplace/ui-components-library/datasources/datasource-inline.html
+---
+
+This document explains the Datasource Dependable service in the Components Library.
+
+## Overview
+
+Datasource Dependable is an Angular service that resolves datasources for a component based on the value of a specific element.
+
+Check out an example usage of the Datasource Dependable.
```
[the remark this answers is not in the document]
Comment (danilmoro3) [reply docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-dependable.md:22]: Only first one is an example of usage

File: docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger-change.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+Check out an example usage of the Datasource Trigger Change.
+
+Service configuration:
+
+| ATTRIBUTE | DESCRIPTION |
+| - | - |
+|type |  A datasource type. |
+|event |  An event type triggered by element. |
+|debounce |  Delays the emission of values the next datasource (default is `300ms`). |
```
Comment (andriitserkovnyi) [inline docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger-change.md:30]: Can you clarify the description? Looks like something is missing

File: docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    }
+    ...
+}
+```
+
+## Main service
+
+The main module lets you register any `datasource.trigger` by key using the `withEvents()` static method. It assigns the object of datasources to `DatasourceEventTypesToken` under the hood.
+
+The `resolve()` method gets the trigger element using the `DatasourceTriggerElement` abstract class (token), finds a specific service from `DatasourceEventTypesToken` by `config.event` (from the argument) and returns observable with data (based on the data returned from trigger element) by `Datasource.resolve()`.
```
Comment (andriitserkovnyi) [inline docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger.md:61]: Split this into several sentences and try to remove the braces

File: docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    ],
+})
+export class RootModule {}
+```
+
+The context within which Datasources operate is defined by the local injector where it’s being used.
+
+## Interfaces
+
+Interfaces for the Datasource Trigger configuration and types:
```
Comment (andriitserkovnyi) [inline docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger.md:111]: Types of what?

Comment (danilmoro3) [reply docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger.md:111]: Interfaces and types for the Datasource Trigger configuration:

File: docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger-change.md (hunk 2 above)
Comment (danilmoro3) [reply docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger-change.md:28]: What exactly? Description looks clear

File: docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger.md (hunk 3 above)
Comment (danilmoro3) [reply docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger.md:61]: In this case context will be missed

File: docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger-change.md (hunk 2 above)
Comment (andriitserkovnyi) [reply docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger-change.md:28]: I don't get how `the next datasource` relates to the preceding part of the sentence

Comment (danilmoro3) [reply docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger-change.md:28]: `datasource` is an angular service and not exactly `data source`

Comment (andriitserkovnyi) [reply docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger-change.md:28]: Still don't get it, but I'll try to guess: 
```suggestion
|debounce |  Delays the emission of values of the next datasource (default is `300ms`). |
```

**Pull request merged**